analytics/private-cloud/quotaChanges.ts

```diff
@@ -0,0 +1,95 @@
+import { PrivateCloudRequestedProject, DecisionStatus } from '@prisma/client';
+import prisma from '@/lib/prisma';
+
+export type DataPoint = {
+  date: string;
+  'Quota requests': number;
+};
+
+function createMonthKey(date: Date) {
+  const formatter = new Intl.DateTimeFormat('en', { month: 'short', year: '2-digit' });
+  return formatter.format(date);
+}
+
+export function isQuotaChanged(
+  projectOne: PrivateCloudRequestedProject,
+  projectTwo: PrivateCloudRequestedProject,
+): boolean {
+  // Assuming productionQuota, testQuota, developmentQuota, and toolsQuota are defined and comparable
+  return !(
+    JSON.stringify(projectOne.productionQuota) === JSON.stringify(projectTwo.productionQuota) &&
+    JSON.stringify(projectOne.testQuota) === JSON.stringify(projectTwo.testQuota) &&
+    JSON.stringify(projectOne.developmentQuota) === JSON.stringify(projectTwo.developmentQuota) &&
+    JSON.stringify(projectOne.toolsQuota) === JSON.stringify(projectTwo.toolsQuota)
+  );
+}
+
+export function sortDatesIntoMonths(dates: Date[]) {
+  const monthlyChanges: { [key: string]: Date[] } = {};
+  dates.forEach((date) => {
+    const monthKey = createMonthKey(date);
+    if (!monthlyChanges[monthKey]) {
+      monthlyChanges[monthKey] = [];
+    }
+    monthlyChanges[monthKey].push(date);
+  });
+  return monthlyChanges;
+}
+
+export async function detectQuotaChangesByMonth(licencePlate: string, decisionStatus?: DecisionStatus) {
+  const requests = await prisma.privateCloudRequest.findMany({
+    where: { licencePlate, decisionStatus },
+    orderBy: { created: 'asc' },
+
```

Comment:
if `requestedProject` is the only field required for the logic, we can add it to the query like:
```
{
...
select: { requestedProject: true },
...
}